addons/area_markers/functions/fnc_module.sqf

```diff
@@ -0,0 +1,29 @@
+#include "script_component.hpp"
+/*
+ * Author: mharis001
+ * Zeus module function to create an area marker.
+ *
+ * Arguments:
+ * 0: Logic <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [LOGIC] call zen_area_markers_fnc_module
```

Comment:
Seems a bit weird to even have a module. I think that the context menu would suffice or if possible double click on the map could be even better.

Reply:
Not everyone uses the context menu, it can be disabled.

Reply:
But a double click on the map seems a better alternative than a context-based module that takes up space in the tree.

Reply:
Out of Zeus, if you double click on the map it creates a new marker and in Zeus you could use the same behaviour (only you create an area marker, perhaps could be extended later to allow all kinds of markers?). As for the attributes that would be only valid if we would be double-clicking on an object but in this instance that isn't the case.

Reply:
Attributes are for all Zeus entities (objects, groups, waypoints, markers). There is no method that I know of to create a Zeus editable marker.

I just don't see the need to make this accessible via double click when it's available through a right-click + left-click (context menu). Especially when it is such a niche feature.

Reply:
The issue is that the module is profoundly context-sensitive (e.g. you need the map to be open), whereas all the other modules are more about asking `place me on an object`.
And as the project evolves, the more of these modules you have, the bigger mess you are going to have of the module tree, especially if they are such a niche item as you said. What are you going to keep doing down the line? Add more of these types of modules? It doesn't scale.

Achilles has a module filter, but, I personally, never used it because it was too much of a hassle to use, especially if you just needed that module that was locked away. So that too isn't a viable solution to remove the module if you don't want it.

You already provide the option to edit and delete via the map without using the context menu then it would seem completely intuitive that you could create the marker via double click as well.

I'm not talking about adding attributes to the area markers, you're talking about adding something to the mix that isn't required right now.

I'm not saying that this should be restricted to only area markers. It could be later extended to allow usage of any markers not just area ones. Then you wouldn't have to deal with using that marker tree. This would drastically boost its potential later.

Reply:
> Then you wouldn't have to deal with using that marker tree.

The markers tree is the easiest way to create a marker (especially after the integration of the `markers_tree` component), I don't see how any other solution would be more straight forward. And in either case, there is no way to create a Zeus editable marker through script.

Any properties for the regular (icon) Zeus markers would be go through attributes, it does not make sense to have all other Zeus entity types be configured through attributes but markers through something separate. Right now, this UI is specific to area markers (shape, brush), trying to accommodate into one or selectively display elements does not sound like a good idea.

> I'm not talking about adding attributes to the area markers, you're talking about adding something to the mix that isn't required right now.

That is not what I said. I said that attributes are for "regular" Zeus entities and area markers are not one them.

> The issue is that the module is profoundly context-sensitive (e.g. you need the map to be open), whereas all the other modules are more about asking place me on an object.

The module needs to be placed on the map because that is what makes. It could easily be made to just take the position of the module when not on the map. However, that doesn't make sense - markers are a map related entity and it makes sense for interactions with them to go through the map.

> And as the project evolves, the more of these modules you have, the bigger mess you are going to have of the module tree, especially if they are such a niche item as you said. What are you going to keep doing down the line? Add more of these types of modules? It doesn't scale.

I don't see the module tree becoming a mess as long as each module provides a unique feature and is logically categorized. The number of items in the tree will grow as features are added, the only way you avoid that is through the exclusion of said features.

> You already provide the option to edit and delete via the map without using the context menu then it would seem completely intuitive that you could create the marker via double click as well.

I get what you are saying - on the regular "infantry" map, you create markers by double-clicking and I can see an argument for mimicking that on the Zeus map. However, this is only for area markers, as I said earlier, we can't create the icon Zeus markers through script. And in that case, for just area markers, the context menu is a simple out-of-the-way solution to provide this functionality (and the module if someone has the context menu disabled). We also currently have double-click to edit, and CTRL + double-click to remote control.

This is something that can always be changed/integrated later based on feedback. Personally, I don't see a need for it.